SCClassLibrary/Common/Control/Server.sc

```diff
@@ -408,18 +406,16 @@ Server {
 			this.bootInit;
 			if (Server.postingBootInfo) { "prRun: % - %\n".postf(this, "ServerBoot.run") };
 			ServerBoot.run(this);
-			0.2.wait;
 			this.sync;
 			if (Server.postingBootInfo) { "prRun: % .%\n".postf(this, "initTree") };
 			this.initTree;
-			0.2.wait;
 			this.sync;
+			this.bootStatus_(\running); // or maybe 'running'
+			// also set server.statusWatcher.serverRunning_(true) here;
 			if (Server.postingBootInfo) { "prRun: % .%\n".postf(this, "tempBootItems.do") };
-			tempBootItems.do(_.value);
-			tempBootItems.clear;
-			0.2.wait;
-			this.sync;
-			this.bootStatus_(\running);
+			while { tempBootItems.notNil } {
```

Comment:
`tempBootItems` is always notNil -- therefore, this loop ends when it tries to do `removeAt(0)` on an empty List and throws an error.

```
Shared memory server interface initialized
localhost: bootStatus is now 'running'.
ERROR: Primitive '_BasicRemoveAt' failed.
Index out of range.
```

You wanted `notEmpty`.

A quick test before committing/pushing would have identified this issue. We have a bad habit in SC-land of: "That should work -- commit and push." But it really should be: "Let me try it first." It should *always* be "Let me try it first."

Reply:
Yes, you are right - I tested right after committing, saw it and fixed it, but only locally :-(
I do try to improve my habits though ...

Reply:
fixed now.